alicloud/resource_alicloud_slb_domain_extension.go

```diff
@@ -0,0 +1,155 @@
+package alicloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/slb"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudSlbDomainExtension() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunSlbDomainExtensionCreate,
+		Read:   resourceAliyunSlbDomainExtensionRead,
+		Update: resourceAliyunSlbDomainExtensionUpdate,
+		Delete: resourceAliyunSlbDomainExtensionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"load_balancer_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"frontend_port": {
+				Type:         schema.TypeInt,
+				ValidateFunc: validateIntegerInRange(1, 65535),
+				Required:     true,
+				ForceNew:     true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"server_certificate_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAliyunSlbDomainExtensionCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	req := slb.CreateCreateDomainExtensionRequest()
+	req.LoadBalancerId = d.Get("load_balancer_id").(string)
+	req.ListenerPort = requests.NewInteger(d.Get("frontend_port").(int))
+	req.Domain = d.Get("domain").(string)
+	req.ServerCertificateId = d.Get("server_certificate_id").(string)
+	err := resource.Retry(3*time.Minute, func() *resource.RetryError {
```

Comment:
Please refer to the retry in the `resource_alicloud_cen_route_entry.go`, modified the way to return the error and addDebug/SetId.

[https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_cen_route_entry.go](url)